src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateService.scala

```diff
@@ -58,4 +64,33 @@ class JobTemplateServiceImpl(override val jobTemplateRepository: JobTemplateRepo
   override def searchJobTemplates(searchRequest: TableSearchRequest)(implicit ec: ExecutionContext): Future[TableSearchResponse[JobTemplate]] = {
     jobTemplateRepository.searchJobTemplates(searchRequest)
   }
+
+  override def createJobTemplate(jobTemplate: JobTemplate)(implicit ec: ExecutionContext): Future[JobTemplate] = {
+    val userName = getUserName.apply()
+    for {
+      _ <- jobTemplateValidationService.validate(jobTemplate)
+      jobTemplateId <- jobTemplateRepository.insertJobTemplate(jobTemplate, userName)
+    } yield {
+      jobTemplate.copy(id = jobTemplateId)
+    }
+  }
+
+  override def updateJobTemplate(jobTemplate: JobTemplate)(implicit ec: ExecutionContext): Future[JobTemplate] = {
+    val userName = getUserName.apply()
+    for {
+      _ <- jobTemplateValidationService.validate(jobTemplate)
+      _ <- jobTemplateRepository.updateJobTemplate(jobTemplate, userName)
+    } yield {
+      jobTemplate
+    }
+  }
+
+  override def deleteJobTemplate(id: Long)(implicit ec: ExecutionContext): Future[Boolean] = {
+    val userName = getUserName.apply()
+    jobTemplateRepository.deleteJobTemplate(id, userName).map(_ => true)
+  }
+
+  private[services] def getUserName: () => String = {
```

Comment:
This should probably be extracted to a util or service

Reply:
Yes, done